x-pack/plugins/dashboard_enhanced/public/services/drilldowns/abstract_dashboard_drilldown/abstract_dashboard_drilldown.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { DataPublicPluginStart } from 'src/plugins/data/public';
+import { DashboardStart } from 'src/plugins/dashboard/public';
+import { reactToUiComponent } from '../../../../../../../src/plugins/kibana_react/public';
+import {
+  TriggerId,
+  TriggerContextMapping,
+} from '../../../../../../../src/plugins/ui_actions/public';
+import { CollectConfigContainer } from './components';
+import {
+  UiActionsEnhancedDrilldownDefinition as Drilldown,
+  UiActionsEnhancedBaseActionFactoryContext as BaseActionFactoryContext,
+  AdvancedUiActionsStart,
+} from '../../../../../ui_actions_enhanced/public';
+import { txtGoToDashboard } from './i18n';
+import {
+  StartServicesGetter,
+  CollectConfigProps,
+} from '../../../../../../../src/plugins/kibana_utils/public';
+import { KibanaURL } from '../../../../../../../src/plugins/share/public';
+import { Config } from './types';
+
+export interface Params {
+  start: StartServicesGetter<{
+    uiActionsEnhanced: AdvancedUiActionsStart;
+    data: DataPublicPluginStart;
+    dashboard: DashboardStart;
+  }>;
+}
+
+export abstract class AbstractDashboardDrilldown<T extends TriggerId>
+  implements Drilldown<Config, T, BaseActionFactoryContext<T>> {
+  constructor(protected readonly params: Params) {}
+
+  public abstract readonly id: string;
+
+  public abstract readonly supportedTriggers: () => T[];
+
+  protected abstract getURL(config: Config, context: TriggerContextMapping[T]): Promise<KibanaURL>;
+
+  public readonly order = 100;
+
+  public readonly getDisplayName = () => txtGoToDashboard;
+
+  public readonly euiIcon = 'dashboardApp';
+
+  private readonly ReactCollectConfig: React.FC<
+    CollectConfigProps<Config, BaseActionFactoryContext<T>>
+  > = (props) => <CollectConfigContainer {...props} params={this.params} />;
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
```

Comment:
Should we conditionally display filters / time range inside collect config depends if it available on embeddable input? 

These inputs don't make sense in example plugin:
![Screenshot 2020-09-30 at 12 26 21](https://user-images.githubusercontent.com/7784120/94674704-e02bd700-0318-11eb-9993-5d23fbe83780.png)

I am OK if we at least create an issue to clean this up.

Reply:
Good point, I've created an issue for it: https://github.com/elastic/kibana/issues/78997